### ISSUE TITLE
Error in dartium on literal object without named parameters

### ISSUE THREAD

**Issue by a14n** [dart-lang/sdk#24780]
In dartium 1.13.0-dev.7.4 I get an error but it works in dart2js

```
import 'dart:html';
import 'package:js/js.dart';

@JS()
class B {
  external factory B();
}

main() {
  window.console.dir(new B());
}
```

Dartium does not seems to handle correctly an `external factory` without named parameters. The patch generated is :

```dart
patch class B {
  factory B({}) {
    var ret = new js_library.JsObject.jsify({});
    return ret;
  }
}
```

As you can see `factory B({})` is not a valid syntax.

**Comment by kevmoo** [comment]
@jacob314 Can we figure out ASAP if this is critical for 1.13? Looks like @a14n may have a patch...

**Comment by jacob314** [comment]
This should be a one line fix to dartium. Not sure how critical this is as
having a typed interop class with zero properties doesn't seem that useful
in general.
You could always add an optional parameter that you just don't use if you
really want to create a typed interop class with zero properties.

**Comment by kevmoo** [comment]
@jacob314 that work-around seems very hard to document.

@a14n Are you blocked on this at all?

**Comment by a14n** [comment]
Not blocked. It was just the first test I did : anonymous object with empty constructor and getters/setters.

**Comment by jacob314** [comment]
CL containing a test sent out for review.
https://codereview.chromium.org/1431513004/

**Comment by mit-mit** [comment]
Closing as the CL was landed. Do we need to cherry pick this?

**Comment by kevmoo** [comment]
Cherry-pick request already live - https://github.com/dart-lang/sdk/issues/24802

**Comment by jacob314** [comment]
Fix landed and merged into dev.
